lib/pure/strutils.nim

```diff
@@ -844,10 +844,12 @@ proc split*(s: string, sep: string, maxsplit: int = -1): seq[string] {.noSideEff
     doAssert "a,b,c".split(",") == @["a", "b", "c"]
     doAssert "a man a plan a canal panama".split("a ") == @["", "man ", "plan ", "canal panama"]
     doAssert "".split("Elon Musk") == @[""]
+    doAssert "Elon Musk".split("") == @["Elon Musk"]
     doAssert "a  largely    spaced sentence".split(" ") == @["a", "", "largely", "", "", "", "spaced", "sentence"]
 
     doAssert "a  largely    spaced sentence".split(" ", maxsplit=1) == @["a", " largely    spaced sentence"]
-  doAssert(sep.len > 0)
+  if sep.len == 0:
```

Comment:
perform this check in ``splitCommon``, also make sure that ``rsplit`` has the same.